### PR TITLE
Update Rizin to the latest dev

### PR DESCRIPTION
**Pull request by XVilka**
**Your checklist for this pull request**
- [ ] I've read the [guidelines for contributing](https://cutter.re/docs/contributing/code/getting-started.html) to this repository
- [ ] I made sure to follow the project's [coding style](https://cutter.re/docs/contributing/code/development-guidelines.html)
- [ ] I've updated the [documentation](https://cutter.re/docs/user-docs.html) with the relevant information (if needed)

**Detailed description**

Use the new API of Rizin

**Test plan (required)**

**Closing issues**

### CODE REVIEW

File: src/core/Cutter.cpp [hunk 1]
```diff
@@ -722,11 +722,11 @@ PRzAnalysisBytes CutterCore::getRzAnalysisBytesSingle(RVA addr)
     rz_io_read_at(core->io, addr, buf, sizeof(buf));
 
     auto seek = seekTemp(addr);
-    auto vec = fromOwned(rz_core_analysis_bytes(core, addr, buf, sizeof(buf), 1));
+    auto abiter = rz_core_analysis_bytes(core, addr, buf, sizeof(buf), 1);
+	auto ab = abiter
+			? reinterpret_cast<RzAnalysisBytes *>(rz_iterator_next(abiter))
+			: nullptr;
```
Comment (wargio) [inline src/core/Cutter.cpp]: this leaks abiter

File: src/core/Cutter.cpp [hunk 2]
```diff
@@ -1027,14 +1026,14 @@ RVA CutterCore::nextOpAddr(RVA startAddr, int count)
...
+    auto abiter = fromOwned(rz_core_analysis_bytes(core, core->offset, core->block,
+                                                   (int)core->blocksize, count + 1));
 
     RVA addr = startAddr + 1;
-    if (!vec) {
+    if (!abiter) {
         return addr;
     }
-    auto ab = reinterpret_cast<RzAnalysisBytes *>(rz_pvector_tail(vec.get()));
+    auto ab = reinterpret_cast<RzAnalysisBytes *>(rz_iterator_next(abiter.get()));
```
Comment (XVilka) [inline src/core/Cutter.cpp]: I just noticed my mistake here - it should be the end of the iterator. We need a new API in Rizin - `rz_iterator_last()` or something like that?

Comment (wargio) [reply src/core/Cutter.cpp]: yes. we do need this.

**Review by wargio — CHANGES_REQUESTED** [review]

File: src/core/Cutter.cpp (hunk 2 above)
Comment (XVilka) [reply src/core/Cutter.cpp]: Added a new API here: https://github.com/rizinorg/rizin/pull/4278

**Review by b1llow — APPROVED** [review]

**Pull request merged**